### PR TITLE
Update saml inline hook doc to add SessionNotOnOrAfter support

### PR DESCRIPTION
**Pull request by nakuljoshi-okta**
## Description:
- Update SAML inline hook doc to add `SessionNotOnOrAfter` support. Okta now allows clients to add `sessionLifetime` using inline hook command which will calculate `SessionNotOnOrAfter` value and return it in SAML assertion response. 
More details: [OKTA-353804](https://oktainc.atlassian.net/browse/OKTA-353804)

- **Is this PR related to a Monolith release?** Yes. Monthly Release: 2021.05.0 , Date: 4/30/2021

### Resolves:

* [OKTA-380909](https://oktainc.atlassian.net/browse/OKTA-380909)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/reference/saml-hook/index.md [hunk 1]
```diff
@@ -334,14 +334,18 @@ The `value` object is where you specify the specific operation to perform. It is
 
 Specify the location within the assertion where you want to apply your operation using a slash-delimited path, which follows JSON Patch conventions.
 
-When you perform an `add` op to add a new attribute statement, always begin with `/claims/` and follow that with the name of the new attribute that you are adding.
+When you perform an `add` op to add a new attribute statement, begin with `/claims/` and follow that with the name of the new attribute that you are adding.
```
Comment (brianduffield-okta) [inline packages/@okta/vuepress-site/docs/reference/saml-hook/index.md:337]: Should we add a sentence about the ability to add to the `/authentication` path? Or is this a limited scenario?

File: packages/@okta/vuepress-site/docs/reference/saml-hook/index.md [hunk 2]
```diff
@@ -334,14 +334,18 @@ The `value` object is where you specify the specific operation to perform. It is
...
 
 When you modify an existing assertions statement, begin the path with `/subject/`, `/authentication/`, `/conditions/`, or `/claims/`, depending on which part of the assertion you want to modify. You then drill down within the child elements using slash-delimited element names, for example, `/claims/array/attributeValues/1/value`. (The `/1/` in the path indicates the index of the array.)
 
 ### URI claims
 
 Okta supports URI claims with SAML assertion hooks. When you need to replace or add a URI claim, you must encode the claim name within the command per the [JavaScript Object Notation (JSON) Pointer](https://tools.ietf.org/html/rfc6901) specification. Specifically, this replaces `~` with `~0` and  `/` with `~1`.
 
+### SessionNotOnOrAfter support
+
+If you want to add `SessionNotOnOrAfter` attribute to the `<saml:AuthnStatement>` in SAML assertion, use `add` op with the path `/authentication/sessionLifetime` and session lifetime in seconds. Refer to example [here](/docs/reference/saml-hook/#sample-listing-of-json-payload-of-response). Okta will calculate `SessionNotOnOrAfter` by adding `/authentication/sessionLifetime` value to `issueInstant` and return it in SAML assertion.
```
Comment (brianduffield-okta) [inline packages/@okta/vuepress-site/docs/reference/saml-hook/index.md:347]:
```suggestion
In some scenarios, your service provider may require the `SessionNotOnOrAfter` attribute for the `<saml:AuthnStatement>` in the SAML assertion, which sets the provider session time correctly. Use `add` op with the path `/authentication/sessionLifetime` and a value for session lifetime in seconds to add this attribute. See the [Sample listing of JSON payload response](/docs/reference/saml-hook/#sample-listing-of-json-payload-of-response) for an example. Okta calculates `SessionNotOnOrAfter` by adding `/authentication/sessionLifetime` value to the `issueInstant` attribute and returns it in the SAML assertion.
```

Comment (brianduffield-okta) [reply packages/@okta/vuepress-site/docs/reference/saml-hook/index.md:347]: @nakuljoshi-okta fyi, hope you don't mind a bit of a re-write. I only add a reason why I believe you would want to update this attribute in first sentence (only add if it's accurate). I then reformatted a few items for dev docs styles.

File: packages/@okta/vuepress-site/docs/reference/saml-hook/index.md (hunk 1 above)
Comment (nakuljoshi-okta) [reply packages/@okta/vuepress-site/docs/reference/saml-hook/index.md:337]: Yup this is a limited scenario. It only allows `/authentication/sessionLifetime` which is the special case.

File: packages/@okta/vuepress-site/docs/reference/saml-hook/index.md (hunk 2 above)
Comment (nakuljoshi-okta) [reply packages/@okta/vuepress-site/docs/reference/saml-hook/index.md:347]: this re-write looks better to me. committing the suggestion.

**Review by brianduffield-okta — APPROVED** [review]
LGTM

**Pull request merged**